### PR TITLE
LA-201: Add a new Select Product endpoint

### PR DESCRIPTION
**Pull request by bpoon-selina**

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **bpoon-selina**
You can assign the PR to them by writing `/assign @bpoon-selina` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
@bpoon-selina: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | c063c7fc79019d9db9d0ac8a6780e483566dcba2 | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-95-pr-vx4sj) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-95** [here](http://ms-lending-api-pr95.selinadev.co.uk)

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-95** [here](http://ms-lending-api-pr95.selinadev.co.uk)

File: src/main/java/com/selina/lending/api/controller/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public ResponseEntity<SelectProductResponse> selectProductOffer(String externalApplicationId, String productCode) {
+        log.info("Select product for [externalApplicationId={}] [productCode={}]", externalApplicationId, productCode);
+        var response = SelectProductResponseMapper.INSTANCE.mapToSelectProductResponseDto(productService.selectProductOffer(externalApplicationId, productCode));
+        enrichResponse(response, externalApplicationId, productCode);
+        return ResponseEntity.ok(response);
+    }
+
+    private void enrichResponse(SelectProductResponse response, String externalApplicationId, String productCode) {
```
Comment (Vadym-Lopatka) [inline src/main/java/com/selina/lending/api/controller/ProductController.java:47]: does not look like controller responsibility, more related to a mapper

File: src/main/java/com/selina/lending/internal/service/ProductServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    public ProductServiceImpl(MiddlewareRepository middlewareRepository, MiddlewareApplicationServiceRepository middlewareApplicationServiceRepository, AccessManagementService accessManagementService) {
+        this.middlewareRepository = middlewareRepository;
+        this.middlewareApplicationServiceRepository = middlewareApplicationServiceRepository;
+        this.accessManagementService = accessManagementService;
+    }
+
+    @Override
+    public SelectProductResponse selectProductOffer(String externalApplicationId, String productCode) {
+        var applicationIdentifier = middlewareApplicationServiceRepository.getApplicationIdByExternalApplicationId(externalApplicationId);
```
Comment (Vadym-Lopatka) [inline src/main/java/com/selina/lending/internal/service/ProductServiceImpl.java:45]: looks like this method would be safer to invoke after the check of access
in this way, when access will be denied - we do not do an unnecessary remote call

**Review by Vadym-Lopatka — CHANGES_REQUESTED** [review]

File: src/main/java/com/selina/lending/internal/service/ProductServiceImpl.java (hunk 2 above)
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/internal/service/ProductServiceImpl.java:45]: ok, makes sense

File: src/main/java/com/selina/lending/api/controller/ProductController.java (hunk 1 above)
Comment (bpoon-selina) [reply src/main/java/com/selina/lending/api/controller/ProductController.java]: done

**Review by Vadym-Lopatka — APPROVED** [review]

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[Vadym-Lopatka](https://github.com/Selina-Finance/ms-lending-api/pull/95#pullrequestreview-1194034907 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)~~ [Vadym-Lopatka]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-95** [here](http://ms-lending-api-pr95.selinadev.co.uk)

**Pull request merged**